### ISSUE TITLE
clients.NewAviClient raised a nil point panic when the avi-controller host can not be accessed

### ISSUE THREAD

**Issue by caidongqing** [vmware/alb-sdk#2685]
### Describe the bug

the clients.NewAviClient function will raised a nil point panic if the avi-controller host can not be accessed. when I specify a wrong avi-controller host ip, I encounter the below error message:

```
goroutine 566 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
 /go/pkg/mod/sigs.k8s.io/[controller-runtime@v0.12.3](mailto:controller-runtime@v0.12.3)/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x2455b20, 0x440dbc0})
 /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/vmware/alb-sdk/go/session.(*AviSession).restRequest(0xc0015fa000, {0x29639f8, 0x4}, {0x2965463, 0x5}, {0x2437800, 0xc0030e5950}, {0x0, 0x0}, {0x0, ...}, ...)
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/session/avisession.go:713 +0x52b
github.com/vmware/alb-sdk/go/session.(*AviSession).restRequestInterfaceResponse(0xc001b88b01?, {0x29639f8, 0x4}, {0x2965463, 0x5}, {0x2437800, 0xc0030e5950}, {0x0, 0x0}, {0x0, ...})
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/session/avisession.go:1108 +0x17b
github.com/vmware/alb-sdk/go/session.(*AviSession).Post(0x8?, {0x2965463?, 0x30?}, {0x2437800?, 0xc0030e5950?}, {0x0?, 0x0?}, {0x0, 0x0, 0x0})
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/session/avisession.go:1131 +0x85
github.com/vmware/alb-sdk/go/session.(*AviSession).initiateSession(0xc0015fa000)
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/session/avisession.go:390 +0x2b8
github.com/vmware/alb-sdk/go/session.NewAviSession({0xc001b9f930, 0xe}, {0xc001b9f948, 0x5}, {0xc001392dc0, 0x4, 0x20?})
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/session/avisession.go:350 +0x3a9
github.com/vmware/alb-sdk/go/clients.NewAviClient({0xc001b9f930, 0xe}, {0xc001b9f948, 0x5}, {0xc001392dc0, 0x4, 0x4})
 /go/pkg/mod/github.com/vmware/[alb-sdk@v0.0.0-20230202142453-a9968b9cd48d](mailto:alb-sdk@v0.0.0-20230202142453-a9968b9cd48d)/go/clients/avi_client.go:164 +0x85
```

### Reproduction steps

It's easy to reproduce, such as the demo-code below:

```
package main

 import ( 
     "fmt"
     cli "github.com/vmware/alb-sdk/go/clients"
     "github.com/vmware/alb-sdk/go/session"
 )

 const (
    AviControllerRetryCount    = 1
    AviControllerRetryInterval = 1
)
 
 func main() {
    options := []func(*session.AviSession) error{
        session.SetPassword("your_passwork"),
        session.SetControllerStatusCheckLimits(AviControllerRetryCount, AviControllerRetryInterval),
    }
    
    options = append(options, session.SetInsecure)

    _, err := cli.NewAviClient("avi_controller_ip", "avi_user", options...)
    if err != nil {
        fmt.Println("error occurred, it's not the panic which can be caught here...")
    }
}
```

### Expected behavior

The nil point runtime panic should be handled in alb-sdk, although maybe we can catch the panic and handle it in local project.

**Comment by parimanur** [comment]
@caidongqing Can you add the details of the SDK version and the error seen in the test code? 
I do not see the error when I pass a host which is unreachable. 

Log messages when the panic was seen also will help.
cc: @Rohan-sss1

**Comment by Rohan-sss1** [comment]
@caidongqing we already fixed this issue, could you try it on latest 22.1.4 and update us.

**Comment by caidongqing** [comment]
Hi @parimanur and @Rohan-sss1 Thanks for your prompt reply. 
I have provided the log messages in the "Describe the bug" section, and the version could be seen in the log, it's alb-sdk@v0.0.0-20230202142453-a9968b9cd48d, I checked the commit history, it corresponds to version 30.1.1 

@Rohan-sss1 I have tried the latest version 22.1.4 and it does indeed fixed this issue, thanks.
In addition, I'd like to confirm which commit fixed this issue, is it included in version 30.1.1?

thanks
caidognqing
